### PR TITLE
Create webhook to provision runners on AWS

### PR DESCRIPTION
**Pull request by aosoriodw**
Create an HTTP server that can received GitHub's webhook requests and create a Mac M1 runner in response to them.

**Tests**:
- Tested by pushing a change to GitHub and verifying in the logs that the action got the request and tried to provision a machine.
- Other manual tests to ensure that errors are handled correcly

### CODE REVIEW

File: github-runner-provisioner/macm1.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	placement            types.Placement
+	instanceCount        int32
+	shutdownBehavior     types.ShutdownBehavior
+	instanceType         types.InstanceType
+}
+
+var macM1HostResourceGroupArn = "arn:aws:resource-groups:us-east-1:914373874199:group/GitHub-Runners"
+
+var macM1Config = runnerConfig{
+	imageId: "ami-01b8fcd5770ceb9c1",
```
Comment (AidanWenzel) [inline github-runner-provisioner/macm1.go:22]: We should add a comment here about what this AMI is (macOS 12.6 on arm64)

File: github-runner-provisioner/macm1.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	instanceType:     "mac2.metal",
+}
+
+func macRunnerUserData(ctx context.Context, owner string, repo string) (string, error) {
+	token, err := getGitHubRunnerToken(ctx, owner, repo)
+	if err != nil {
+		return "", err
+	}
+
+	const labels = "macOS-arm64"
```
Comment (AidanWenzel) [inline github-runner-provisioner/macm1.go:37]: I wonder if we should put the OS version as a label also? I could see a world where we want to test on 12.5, 12.4 etc

**Review by AidanWenzel — CHANGES_REQUESTED** [review]

File: github-runner-provisioner/macm1.go (hunk 1 above)
Comment (aosoriodw) [reply github-runner-provisioner/macm1.go]: done

File: github-runner-provisioner/macm1.go (hunk 2 above)
Comment (aosoriodw) [reply github-runner-provisioner/macm1.go:37]: That makes sense, but I'd like to do that on a new PR.
I think we could have options like:
`macOS-arm64-12.5`
`macOS-arm64-latest`

**Review by AidanWenzel — APPROVED** [review]
LGTM!

**Pull request merged**